src/main/java/com/example/hibernate/exception/EntityNotFoundException.java

```diff
@@ -0,0 +1,12 @@
+package com.example.hibernate.exception;
+
+public class EntityNotFoundException extends RuntimeException {
+    public EntityNotFoundException(String entity, Object id) {
+        super("Entity '%s' with id = '%s' not found!".formatted(entity, id));
+    }
+
+    public EntityNotFoundException(String entity, Object id, Object id2) {
+        super("Entity '%s' with id = '%s' & id = '%s' not found!".formatted(entity, id, id2));
```

Comment:
I think you use this for e.g. Part with ID does not exist and Model with ID does not exists, but in reality they should be two throws really (as we spoke on Slack):

```java
Optional<Part> optionalPart = partRepo.getPartById(partId)
if (optionalPart.isEmpty())
     throw new EntityNotFoundException('Part', partId);
Optional<Model> optionalModel = modelRepo.getModelById(modeld)     
if (optionalRepo.isEmpty())
    throw new EntityNotFoundException('Model', modelId)
```